File: RLGG UI实例/其他/先进项目/工具箱关联/UI编辑器/收徒/+容器直装的入门与源码/mix ui/app/src/main/java/com/mucheng/mucuteuix/example/MainActivity.java
package com.mucheng.mucuteuix.example;

import android.app.Activity;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;

import com.mucheng.mucuteuix.example.ui.MIX;

public class MainActivity extends Activity implements Runnable {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    getWindow().getDecorView().postDelayed(this, 350L);
  }

  @Override
  public void run() {
    MIX.init(this);
  }

}

File: RLGG UI实例/其他/先进项目/工具箱关联/UI编辑器/收徒/+容器直装的入门与源码/苏沐橙nodejava/app/src/main/java/com/mojang/minecraftpe/MainActivity.kt
package com.mojang.minecraftpe

import android.app.Activity
import android.os.Bundle
import com.mucheng.nodejava.core.Context
import com.mucheng.nodejava.core.Isolate
import java.io.File
import java.lang.ref.WeakReference

class MainActivity : Activity() {

    companion object {

        @JvmStatic
        lateinit var currentMainActivity: WeakReference<MainActivity>

    }

    private lateinit var isolate: Isolate

    private lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        currentMainActivity = WeakReference(this)
        extraAssets()

        isolate = Isolate()
        context = Context(isolate, filesDir.absolutePath)
        context.injectJavaBridge()
        context.evaluateScript(File(filesDir, "nodeJava.js").readText())
        context.evaluateScript(File(filesDir, "main.js").readText())
        context.spinEventLoopNoWait()
    }

    private fun extraAssets() {
        val baseName = "nodejava"
        val items = assets.list(baseName) ?: emptyArray()
        for (item in items) {
            val fullName = "$baseName/$item"
            val content = readFromAssets(fullName)
            File(filesDir, item).writeText(content)
        }
    }

    private fun readFromAssets(path: String) =
        assets.open(path).bufferedReader().use { it.readText() }

}